great_expectations/checkpoint/checkpoint.py

```diff
@@ -225,8 +230,8 @@ def run(  # noqa: C901, PLR0913, PLR0915
             "expectation_suite_ge_cloud_id": expectation_suite_ge_cloud_id,
         }
 
-        substituted_runtime_config: dict = self.get_substituted_config(
-            runtime_kwargs=runtime_kwargs
+        substituted_runtime_config: dict = self._get_substituted_config(
+            config_provider=context.config_provider, runtime_kwargs=runtime_kwargs
```

Comment:
Aggressively cutting the public surface area of this class - we should only really have `run()`